.github/workflows/pytest.yaml

```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Run PyTest
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Generally we should run tests on all pull request pushes (plus merges, i.e. pushes, to main as you have above).  So we usually leave out any `branches` here.